ibm/service/kubernetes/resource_ibm_container_dedicated_host.go

```diff
@@ -0,0 +1,482 @@
+				return diag.Errorf("[ERROR] DisableDedicatedHostPlacement failed: %v", err)
+			}
+		}
+		_, err = waitForDedicatedHostPlacement(ctx, dedicatedHostAPI, hostID, hostPoolID, placement.(bool), DedicatedHostCreateTimeout, targetEnv)
+		if err != nil {
+			return diag.Errorf("[ERROR] waitForDedicatedHostPlacement failed: %v", err)
+		}
+		d.Set("placement_enabled", placement)
+	}
+
+	return nil
+}
+
+func resourceIBMContainerDedicatedHostRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	id := d.Id()
+	if err := getIBMContainerDedicatedHost(id, d, meta); err != nil {
+		return diag.Errorf("[ERROR] getIBMContainerDedicatedHost failed: %v", err)
+	}
+	return nil
+}
+
+func getIBMContainerDedicatedHost(id string, d *schema.ResourceData, meta interface{}) error {
+	client, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return err
+	}
+	dedicatedHostAPI := client.DedicatedHost()
+	targetEnv := v2.ClusterTargetHeader{}
+
+	// <hostpoolid>:<hostid>
+	m := strings.Split(id, ":")
+	if len(m) < 2 || m[0] == "" || m[1] == "" {
+		return fmt.Errorf("[ERROR] unexpected format of ID (%s), the expected format is <hostpoolid>:<hostid>", id)
+	}
+	hostPoolID := m[0]
+	hostID := m[1]
+
+	dedicatedHost, err := dedicatedHostAPI.GetDedicatedHost(hostID, hostPoolID, targetEnv)
+
+	if err != nil {
+		if apiErr, ok := err.(bmxerror.RequestFailure); ok {
+			if apiErr.StatusCode() == 404 {
+				return nil
+			}
+		}
+		return fmt.Errorf("[ERROR] Error getting container dedicatedhost: %s", err)
+	}
+
+	d.Set("host_pool_id", hostPoolID)
+	setDedicatedHostFields(d, dedicatedHost)
+
+	return nil
+}
+
+func resourceIBMContainerDedicatedHostUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(conns.ClientSession).VpcContainerAPI()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	dedicatedHostAPI := client.DedicatedHost()
+	targetEnv := v2.ClusterTargetHeader{}
+
+	hostPoolID := d.Get("host_pool_id").(string)
+	hostID := d.Get("host_id").(string)
+
+	if d.HasChange("placement_enabled") {
+		placement := d.Get("placement_enabled").(bool)
+		req := v2.UpdateDedicatedHostPlacementRequest{
+			HostPoolID: hostPoolID,
+			HostID:     hostID,
+		}
+		if placement {
+			if err = dedicatedHostAPI.EnableDedicatedHostPlacement(req, targetEnv); err != nil {
+				return diag.Errorf("[ERROR] EnableDedicatedHostPlacement failed: %v", err)
+			}
+		} else {
+			if err = dedicatedHostAPI.DisableDedicatedHostPlacement(req, targetEnv); err != nil {
+				return diag.Errorf("[ERROR] DisableDedicatedHostPlacement failed: %v", err)
+			}
+		}
+		_, err = waitForDedicatedHostPlacement(ctx, dedicatedHostAPI, hostID, hostPoolID, placement, DedicatedHostCreateTimeout, targetEnv)
+		if err != nil {
+			return diag.Errorf("[ERROR] waitForDedicatedHostPlacement failed: %v", err)
+		}
+		d.Set("placement_enabled", placement)
+	}
+	return nil
```

Comment:
After update call the Read so it set back the latest values..

Reply:
I don't call the read, because the waitForDedicatedHostXY functions are executing the same GET calls. we should have the latest values at this point